chapter/related-work.md

```diff
@@ -6,6 +6,90 @@ As such, it is important to get a picture of the field of current LSP projects.
 This chapter will survey a varied range of popular language servers, compare common capabilities, and implementation approaches.
 Additionally, this part aims to recognize alternative approaches to the LSP, in the form of legacy protocols, extensible development platforms LSP extensions and the emerging Language Server Index Format.
 
+
+## Previous Approaches
+
+### IDEs
+
+Before the invention of the Language Server Protocol, extensive language support used to be provided by an IDE.
+Yet, the range of officially supported languages remained relatively small [@intellij-supported-languages].
+While integration for popular languages was common, IDE grade support for less popular ones was all but guaranteed and relied mainly on community efforts.
+In fact Eclipse[@eclipse-a-platform,eclipse-www], IntelliJ[@intelliJ], and Visual Studio[@VisualStudio], to this day the most popular IDE choices, Focus on a narrow subset of languages, historically, Java and .NET.
+Additional languages can be integrated by custom (third-party) plugins or derivations of the base platform ([@list-of-eclipse,@jetbrains-all-products]).
+Due to the proprietary nature of some of these products, plugins are not compatible between different platforms.
```

Comment:
As much as I love open source, I think that's a bit unfair. Vim and emacs plugins are not compatible either. I guess having a common interface for plugins, for editors developed in different languages with different architectures and capabilities is just too constraining.

Reply:
Fair enough, its more general than economics.